### ISSUE TITLE
jsxstyle-loader doesn't extract jsxstyle into seperate css file.

### ISSUE THREAD

**Issue by chenxsan** [jsxstyle/jsxstyle#110]
I'm trying to build some React Comonents with `webpack.prod.config.js`, here's part of my `webpack.prod.config.js` file:

```js
const jsRule = {
  test: /\.js$/,
  include: path.resolve(__dirname, 'src'),
  use: [
    {
      loader: 'jsxstyle-loader'
    },
    {
      loader: 'babel-loader',
      options: {
        presets: ['@babel/preset-env', '@babel/preset-react'],
        plugins: ['@babel/plugin-proposal-object-rest-spread']
      }
    }
  ]
}
let config = {
  entry: entries,
  mode: 'production',
  output: {
    path: path.resolve(__dirname, 'build'),
    filename: '[name].[chunkhash].js',
    libraryTarget: 'umd'
  },
  module: {
    rules: [
      jsRule,
      {
        test: /\.css$/,
        use: [
          MiniCssExtractPlugin.loader,
          {
            loader: 'css-loader',
            options: { minimize: true }
          },
          {
            loader: 'postcss-loader',
            options: {
              ident: 'postcss',
              plugins: () => [require('autoprefixer')()]
            }
          }
        ]
      }
    ]
  },
  plugins: [
    new CleanWebpackPlugin(['build']),
    new JsxstyleLoaderPlugin(),
    new MiniCssExtractPlugin({ filename: '[name].[contenthash].css' }),
    new ManifestPlugin({ seed })
  ],
  externals: [nodeExternals()],
  target: 'node'
}
```
You can check the whole file here https://github.com/OkMarvin/january/blob/master/webpack.prod.config.js.

After I run `npx webpack --config webpack.prod.config.js`, those jsxstyle in components are not extracted, they are still inside the js file.

Any suggest?

**Comment by chenxsan** [comment]
According to https://github.com/smyte/jsxstyle/tree/master/packages/jsxstyle-loader#its-not-working-:

> jsxstyle-loader relies on JSX still being around, so make sure it runs before babel-loader does its thing.

I should put `jsxstyle-loader` after `babel-loader` so it would be executed first, here's the updated config:

```js
const jsRule = {
  test: /\.js$/,
  include: path.resolve(__dirname, 'src'),
  use: [
    {
      loader: 'babel-loader',
      options: {
        presets: ['@babel/preset-env', '@babel/preset-react'],
        plugins: ['@babel/plugin-proposal-object-rest-spread']
      }
    },
    {
      loader: 'jsxstyle-loader',
      options: {
        whitelistedModules: [
          require.resolve('./src/constants')
        ]
      }
    }
  ]
}
```
Now I got those errors:

```
ERROR in ./src/templates/singular.js
    Module build failed: TypeError: Cannot read property 'init' of undefined
        at getStaticBindingsForScope (/Users/sam/Documents/githubRepos/@okmarvin/january/node_modules/jsxstyle-loader/utils/ast/getStaticBindingsForScope.js:70:16)
        at /Users/sam/Documents/githubRepos/@okmarvin/january/node_modules/jsxstyle-loader/utils/ast/extractStyles.js:369:39
        at enter (/Users/sam/Documents/githubRepos/@okmarvin/january/node_modules/jsxstyle-loader/utils/ast/extractStyles.js:391:15)
        at NodePath._call (/Users/sam/Documents/githubRepos/@okmarvin/january/node_modules/jsxstyle-loader/node_modules/@babel/traverse/lib/path/context.js:64:19)
```

**Comment by meyer** [comment]
Ah yep, loader order is important. I should probably warn if jsxstyle is detected but no styles are extracted. Loaders in the wrong order is a pretty common configuration mistake.

I’ve got a webpack 4 compatible version of `jsxstyle-loader` I’m about to publish. I’ll let you know when it’s live!

**Comment by chenxsan** [comment]
I tried to add `if (!dec) continue;` in `getStaticBindingsForScope.js`:

<img width="618" alt="image" src="https://user-images.githubusercontent.com/1091472/39559499-46f3de3a-4ec9-11e8-8311-6075e6b707fe.png">

It works great now in my project.

**Comment by meyer** [comment]
Ah yeah, I caught that error when I [converted `jsxstyle-loader` to TypeScript](https://github.com/smyte/jsxstyle/blob/08ccf7c/packages/jsxstyle-loader/src/utils/ast/getStaticBindingsForScope.ts#L98-L100). It’ll be fixed in the next release.

Glad it’s working well for you! Let me know if you have any more questions.

**Comment by meyer** [comment]
@chenxsan I just published `jsxstyle-loader@1.1.0` which contains the above fix.
